docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,394 @@
+# Derive minimal ServiceAccount required for ClusterExtension Installation and Management
+
+OLMv1 does not provide cluster admin privileges by default for installing cluster extensions. This document will explain how to derive the required RBAC configuration for a provided service account"
+
+The ServiceAccount is specified in the ClusterExtension manifest as shown below:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      version: 0.6.0
+  install:
+    namespace: argocd
+    serviceAccount:
+      name: argocd-installer
+```
+
+The initial stable version (v1.0.0) only supports FBC catalogs containing registry+v1 bundles. OLMv1 will not support all OLMv0 content. OLMv1 will only support bundles that meet the following criteria:
+* AllNamespaces install mode is enabled
+* No dependencies on other packages or GVKs
+* No webhooks
+* Does not make use of the OperatorConditions API
+
+### Required RBAC
+
+You can determine the specifics of these permissions by referencing the bundle of the ClusterExtension you want to install. The service account must have the following permissions:
+
+* ClusterRole with all the roles specified in the bundle ClusterServiceVersion. This includes all the
+  rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+* ClusterRole to create and manage CustomResourceDefinitions
+* Update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+* Permissions to create the controller deployment, this corresponds to the rules to manage the
+  deployment defined in the ClusterServiceVersion
+* Permissions to create the other manifest objects, rules to manage any other cluster-scoped resources
+  shipped with the bundle
+* Rules to manage any other namespace-scoped resources 
+* Permissions to create the necessary roles and rolebindings for the controller to be able to perform its job
+* Get, list, watch, update, patch, delete the specific resources that get created/
+
+### Manual process for minimal RBAC creation
```

Comment:
I haven't had the chance to review this PR, and I am not sure if I will have the cycles to do a close review.

As a broader observation, I am having a hard time seeing the value of describing the manual steps for determining RBAC, when we aren't really production-ready yet.

WDYT of providing a couple of generic SAs/role bindings that will cover most use cases until we can come up with tooling to help out?

Doing it the manual way is a pretty frustrating experience. A difficult procedure is not the worst thing in the world if there is real value or a payoff. I don't think users would get much value out of the experience of doing things the hard way, especially when we are in a "try this out and give us feedback" stage.

Am I missing something? Is there a good reason to do it this way?

Reply:
Quick follow up: This isn't meant to be a comment on the writing or the feedback given.

I am wondering if this is the right approach to a complicated workflow topic that will be replaced with a better UX in the near future.

Reply:
rephrased manual rbac